pkg/config/config.go

```diff
@@ -914,9 +914,14 @@ func ResolveSecrets(config Config, origin string) error {
 	return nil
 }
 
-// Avoid log ingestion breaking because of a newline in the API key
-func sanitizeAPIKey(config Config) {
-	config.Set("api_key", strings.TrimSpace(config.GetString("api_key")))
+// SanitizeAPIKey strips newlines and other control characters from a given key.
+func SanitizeAPIKey(config Config, key string) {
+	config.Set(key, Sanitize(config.GetString(key)))
+}
+
+// Sanitize strips newlines and other control characters from a given string.
+func Sanitize(s string) string {
```

Comment:
`Sanitize` is quite a generic name. 
What do you think about renaming `Sanitize` to `SanitizeAPIKey` and the function line 918 can be renamed from `SanitizeAPIKey` to `SanitizeAPIKeyConfig` for example. Feel free to come with other naming.

Reply:
Makes sense to me 👍